BaseApi/V1/Infrastructure/CorrelationMiddleware.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace BaseApi.V1.Infrastructure
```

Comment:
I think this middleware is specific for the Controller/API and it should probably sits under the controller folder. The infrastructure should hold the model classes representing the domain object as it is in the data source.